File: src/main/java/hiiragi283/material/gui/GuiModuleInstaller.java
package hiiragi283.material.gui;

import hiiragi283.material.api.gui.HiiragiGuiButton;
import hiiragi283.material.api.gui.HiiragiGuiContainer;
import hiiragi283.material.container.ContainerModuleInstaller;
import hiiragi283.material.tile.TileEntityModuleInstaller;
import hiiragi283.material.util.HiiragiUtil;
import net.minecraft.client.gui.GuiButton;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.util.ResourceLocation;
import net.minecraft.util.text.TextComponentString;
import org.jetbrains.annotations.NotNull;

public class GuiModuleInstaller extends HiiragiGuiContainer<TileEntityModuleInstaller> {

    public static final ResourceLocation TEXTURE = HiiragiUtil.getLocation("textures/gui/module_installer.png");

    private final EntityPlayer player;

    public GuiModuleInstaller(TileEntityModuleInstaller tile, EntityPlayer player) {
        super(new ContainerModuleInstaller(tile, player));
        this.player = player;
        this.backGround = TEXTURE;
    }

    @Override
    public void initGui() {
        super.initGui();
        addButton(new HiiragiGuiButton(this, 0, 8 + 18 * 4, 36 + 18, this.xSize, this.ySize, 16, 16, TEXTURE));
    }

    @Override
    protected void actionPerformed(@NotNull GuiButton button) {
        if (button.id == 0) {
            player.sendStatusMessage(new TextComponentString("Button Pressed!"), false);
        }
    }

}

File: src/main/kotlin/hiiragi283/material/gui/GuiModuleInstaller.kt
package hiiragi283.material.gui

import hiiragi283.material.api.gui.HiiragiGuiButton
import hiiragi283.material.api.gui.HiiragiGuiContainer
import hiiragi283.material.container.ContainerModuleInstaller
import hiiragi283.material.util.hiiragiLocation
import hiiragi283.material.util.installModule
import net.minecraft.client.gui.GuiButton
import net.minecraft.client.resources.I18n
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.ResourceLocation

class GuiModuleInstaller(player: EntityPlayer) :
    HiiragiGuiContainer<ContainerModuleInstaller>(ContainerModuleInstaller(player)) {

    override val backGround: ResourceLocation = hiiragiLocation("textures/gui/module_installer.png")

    override fun initGui() {
        super.initGui()
        addButton(HiiragiGuiButton(this, -1, getSlotPosX(5) - 1, getSlotPosY(1) - 1, 18, 18))
    }

    override fun getContainerTitle(): String = I18n.format("tile.module_installer.name")

    override fun drawScreen(mouseX: Int, mouseY: Int, partialTicks: Float) {
        drawDefaultBackground()
        super.drawScreen(mouseX, mouseY, partialTicks)
        renderHoveredToolTip(mouseX, mouseY)
    }

    override fun actionPerformed(button: GuiButton) {
        if (button is HiiragiGuiButton) {
            container.itemHandler.setStackInSlot(
                3,
                installModule(
                    container.itemHandler.getStackInSlot(0),
                    container.itemHandler.getStackInSlot(1),
                    container.itemHandler.getStackInSlot(2)
                )
            )
            //TODO: Send packet to server...?
        }
    }

}